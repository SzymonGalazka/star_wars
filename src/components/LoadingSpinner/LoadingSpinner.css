.SpinnerOverlay {
  height: 100%;
  background: rgba(255,255,255,.6);
  width: 100%;
  left: 0;
  top: 0;
  position: absolute;
  padding-top: 300px;
  opacity: 0.5
}