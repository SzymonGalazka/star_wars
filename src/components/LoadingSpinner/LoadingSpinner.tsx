import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import './LoadingSpinner.css';

const LoadingSpinner = () => {

  return (
    <div className="SpinnerOverlay" data-testid="spinner_overlay">
      <CircularProgress color="secondary" data-testid="spinner" />
    </div>
  );
}

export default LoadingSpinner