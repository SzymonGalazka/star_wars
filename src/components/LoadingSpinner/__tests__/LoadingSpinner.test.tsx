import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, getByTestId } from "@testing-library/react";

import LoadingSpinner from '../LoadingSpinner';

describe("LoadingSpinner", () => {
  it("should render LoadingSpinner component", async () => {
    const tree = TestRenderer.create(<LoadingSpinner />);
    expect(tree).toMatchSnapshot();
  });

  it("should render an overlay div", async () => {
    const { container } = render(
      <LoadingSpinner />
    );
    const element = getByTestId(container, 'spinner_overlay')
    expect(element).toBeTruthy()
  });

  it("should render the spinner", async () => {
    const { container } = render(
      <LoadingSpinner />
    );
    const element = getByTestId(container, 'spinner')
    expect(element).toBeTruthy()
  });

});