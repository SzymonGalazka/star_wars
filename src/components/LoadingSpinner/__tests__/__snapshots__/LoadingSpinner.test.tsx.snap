// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should render LoadingSpinner component 1`] = `
<div
  className="SpinnerOverlay"
  data-testid="spinner_overlay"
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    data-testid="spinner"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
